enhancements/sdk-supports-operands-deletion.md

```diff
@@ -0,0 +1,118 @@
+- The timeout for operand deletion should be user configurable by the SDK
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator author, I want an upstream finalizer no-op handler scaffolded by default and/or library to help write finalizers.
+
+#### Story 2
+
+As an operator author, I want scaffolded fields in the CSV generation by default to enable operand cleanup.
+
+#### Story 3
+
+As an operator author, I would like the Operator SDK's cleanup command to handle deletion of CRDs appropriately based on the flags provided.
+
+### Implementation Details/Notes/Constraints
+
+#### Upstream finalizer no-op handler scaffolded by default and/or library to help write Finalizers
+
+Any operator that handles delete events or if the operator is expecting a delete event for CRs should have Finalizers. The only time Finalizers may not be required is when OwnerReferences exist and they handle all of the cleanup. But as we tend towards cluster scoped operators, OwnerReferences won't be allowed. In that case, all operators should have Finalizers.
+
+To enable this, it would be ideal to have a Finalizer no-op handler scaffolded by default in upstream such as Kubebuilder. The idea is to always scaffold out a Finalizer, a Delete Event Handler and a Reconciler that checks for Finalizers and has a no-op in cases where it's not required to use a Finalizer. 
+
+The Reconciler checks for presence of a Finalizer in a CR object. If the CR object is marked for deletion, which is indicated by the timestamp that was set, the Reconciler runs the finilization logic and removes the finalizer. If the finalization logic fails for some reason, then the Reconciler keeps the Finalizer to retry during the next reconcilation. Typically, the finalization logic would entail necessary cleanup steps that the operator needs to perform before the CR can be deleetd.
+
+Note: If a Finalizer doesn't exist on the CR and the object is marked for delete event, the Reconciler would add a Finalizer to the CR object as part of reconcilcation.
+
+#### Configuration of the cleanup API in SDK
+
+A [ClusterServiceVersion (CSV)][csv] represents a particular version of a running operator on a cluster and is an actual installation of the operator. It's more appropriate to configure operand cleanup API on the CSV than any other resource object.
+
+Configuring the cleanup API elsewhere, such as [Subscription][subscription] for example, will remove the flexibility for a user to opt-in for operand cleanup if the CSV was installed manually without a subscription.
+
+To enable configuration of operand deletion on the CSV, Operator SDK will scaffold the cleanup fields in the CSV spec and enable it to be false by default. The user can then explicitly patch the CSV and set the cleanup field to true if the user would like to opt-in for automatic deletion of operands using OLM. The scaffolding logic of cleanup fields in the CSV may reside in go plugin implementation.
```

Comment:
The `operator generate kustomize manifests` command will add this field to CSVs, not plugin code. CSVs are language-agnostic
```suggestion
To enable configuration of operand deletion on the CSV, Operator SDK will generate CSVs with the `spec.cleanup` field set to `false` by default. The user can then explicitly set this field to true to opt-in to automatic deletion of CRs using OLM.
```